onnx/compose.py

```diff
@@ -0,0 +1,256 @@
+
+    for entry in opset_imports:
+        if entry.domain in opset_import_map:
+            found_version = opset_import_map[entry.domain]
+            if entry.version != found_version:
+                raise ValueError(
+                    "Can't merge two models with different operator set ids for a given domain. "
+                    f"Got: {m1.opset_import} and {m2.opset_import}")
+        else:
+            opset_import_map[entry.domain] = entry.version
+
+    min_ir_version = helper.find_min_ir_version_for(
+        [entry for entry in opset_imports])
+    if ir_version is None:
+        ir_version = max(min_ir_version, max(
+            m1.ir_version, m2.ir_version))
+    if ir_version < min_ir_version:
+        raise ValueError(
+            f"IR version {ir_version} is not sufficient to support "
+            f"the models operator set ids: {opset_imports}")
+
+    graph = merge_graphs(m1.graph, m2.graph, io_map, name, doc_string)
+    model = helper.make_model(graph, producer_name=producer_name,
+                              opset_imports=opset_imports,
+                              ir_version=ir_version)
+    checker.check_model(model)
+    return model
+
+
+def add_prefix(
+        g,  # type: GraphProto
+        prefix,  # type: Text
+        rename_nodes=True,  # type: Optional[bool]
+        rename_edges=True,  # type: Optional[bool]
+        rename_inputs=True,  # type: Optional[bool]
+        rename_outputs=True,  # type: Optional[bool]
+):  # type: (...) -> GraphProto
+    """Adds a prefix to names of elements in a graph: Nodes, Edges, Inputs, Outputs
+
+    It can be used as a utility before merging graphs that have overlapping names.
+    Empty names are not prefixed.
+
+    Arguments:
+        g (GraphProto): Graph
+        prefix (Text): Prefix to be added to each name in the graph
+        rename_nodes (bool): Whether to prefix node names
+        rename_edges (bool): Whether to prefix node edge names
+        rename_inputs (bool): Whether to prefix input names
+        rename_outputs (bool): Whether to prefix output names
+    """
+    if type(g) is not GraphProto:
+        raise ValueError("g argument is not an ONNX graph")
+
+    g = copy.deepcopy(g)
+
+    def prefixed(prefix, name):  # type: (Text, Text) -> Text
+        return prefix + name if len(name) > 0 else name
+
+    if rename_nodes:
+        for n in g.node:
+            n.name = prefixed(prefix, n.name)
+
+    if rename_edges:
+        def clear(lst):  # type: (RepeatedScalarFieldContainer[Text]) -> None
+            for _ in range(len(lst)):
+                lst.pop()
+
+        for n in g.node:
+            out_names = [prefixed(prefix, name) for name in n.output]
+            clear(n.output)
+            n.output.extend(out_names)
+
+            in_names = [prefixed(prefix, name) for name in n.input]
+            clear(n.input)
+            n.input.extend(in_names)
+
+    if rename_inputs:
+        for i in g.input:
```

Comment:
is this enough? even when only rename_inputs is set to true you need to rename the corresponding node inputs right?

Reply:
You are right. I've fixed it (and added tests)